Automattic/wc-calypso-bridge
# Add `wp` as a developer dependency

### Changes proposed in this Pull Request:

This Pull Request introduces an explicit declaration of the `wp` dependency, which improves the reliability of our translation file update process during releases.

In our previous setup, the successful operation hinged on the assumption that developers had `wp` pre-installed on their systems, as suggested by the [installation guide](https://wp-cli.org/#installing). This approach, however, introduced an unnecessary dependency on the individual developer's system environment.

In line with [best practices](https://12factor.net/dependencies), we are now clearly stating that `wp` is required. This new method eliminates potential issues tied to individual system setups and offers a smoother experience for all developers.



Relates to https://github.com/Automattic/i18n-issues/issues/594 .






### How to test the changes in this Pull Request:




1. Execute `npm run prepare-release`, and attempt the first 3 steps
   ```
   1. Bump the version number in the composer.json file and the wc-calypso-bridge.php file.
   2. Update the changelog in readme.txt.
   3. Update the translation files (optional).
   ```
3. Attempt to download the translation files.
6. Verify that the translation files have been updated.
7. Once successful, abort, and delete the branch that was created.
8. Modify the `vendor/bin/wp i18n make-pot` [command](https://github.com/Automattic/wc-calypso-bridge/blob/master/scripts/commands/update-translations.js#L52) to point to a non-existent source directory. e.g Change line 53 of `scripts/commands/update-translations.js` to `vendor/bin/wp i18n make-pot ./non-existent-directory ${ POT_FILE_PATH } --ignore-domain`
9. Attempt steps 1-2.
10. Verify that it failed, and the output terminal describes the error correctly.
11. Delete the branch that was created.



### Other information:




### FOR PR REVIEWER ONLY:
